android/titanium/src/java/org/appcelerator/titanium/view/TiUIView.java

```diff
@@ -1197,6 +1207,14 @@ private void applyCustomBackground(boolean reuseCurrentDrawable)
 		}
 	}
 
+	private void setClipMode(int clipMode)
```

Comment:
I think this should work to get the root view (I haven't tested it though).
```
private void setClipMode(int clipMode)
{
	if (nativeView == null) {
		return;
	}

	View rootView = nativeView;
	while (rootView.getParent() != null && rootView instanceof ViewGroup) {
		rootView = (View) rootView.getParent();
	}

	if (rootView instanceof ViewGroup viewGroup) {
		viewGroup.setClipChildren(clipMode != CLIP_MODE_DISABLED);
	}	
}
```

Reply:
` Error: android.view.ViewRootImpl cannot be cast to android.view.View` think it's one level to high. But I'll do some more tests at another point

Reply:
I think we can remove the `(View)` casting, and declare `rootView` as `ViewParent`, even or simply keep it as an Object as it's already casted in next `if` condition.